### PR TITLE
fix(api): improvements based on using in ci build; docs adds

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: etc/hub.api.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ export class Buckets extends BucketsGrpcClient {
     list(): Promise<Root.AsObject[]>;
     listIpfsPath(path: string): Promise<ListPathItem.AsObject | undefined>;
     listPath(key: string, path: string, depth?: number): Promise<ListPathReply.AsObject>;
-    listPathFlat(key: string, path: string, dirs?: boolean, depth?: number): Promise<ListPathFlat>;
+    listPathFlat(key: string, path: string, dirs?: boolean, depth?: number): Promise<Array<string>>;
```
Comment (andrewxhill) [inline etc/hub.api.md:75]: didn't need a special type obv

File: packages/buckets/src/buckets.spec.ts [hunk 2]
```diff
@@ -145,16 +145,17 @@ describe('Buckets...', () => {
...
-    //   'mybuck/path',
-    //   'mybuck/path/to',
-    //   'mybuck/path/to/file2.jpg'
+    //   '.textileseed',
+    //   'dir1',
+    //   'dir1/file1.jpg',
+    //   'path',
+    //   'path/to',
+    //   'path/to/file2.jpg'
     // ]
```
Comment (andrewxhill) [inline packages/buckets/src/buckets.spec.ts:154]: having the flat paths prefixed by the buck name wasn't helpful when building with this list

File: packages/buckets/src/buckets.ts [hunk 3]
```diff
@@ -77,6 +77,31 @@ const logger = log.getLogger('buckets')
  *     return roots.find((bucket) => bucket.name === bucketName)
  * }
  * ```
+ *
+ * @example
+ * Push an folder in node.js
```
Comment (andrewxhill) [inline packages/buckets/src/buckets.ts:82]: adds streaming file pushes for folder sync via nodejs

File: packages/buckets/src/buckets.ts [hunk 4]
```diff
@@ -348,6 +373,31 @@ export class Buckets extends BucketsGrpcClient {
    *    return await buckets.pushPath(bucketKey!, 'index.html', file)
    * }
    * ```
+   * 
+   * @example
+   * Push an folder in node.js
```
Comment (andrewxhill) [inline packages/buckets/src/buckets.ts:378]: duplicate example. with example checking though, shouldn't be issue keeping updated

File: packages/buckets/src/utils.ts [hunk 5]
```diff
@@ -24,10 +23,10 @@ export async function listPathRecursive(
   path: string,
   depth: number,
   currentDepth = 0,
-) {
+): Promise<ListPathReply.AsObject> {
```
Comment (andrewxhill) [inline packages/buckets/src/utils.ts:26]: better build

File: packages/buckets/src/utils.ts [hunk 6]
```diff
@@ -41,33 +40,36 @@ export async function listPathRecursive(
...
-  // Only push if dirs included or not a dir
-  if (dirs || !tree.isdir) result.push(fp)
-  if (tree.isdir) {
-    for (const item of tree.itemsList) {
-      const downtree = await treeToPaths(item, fp, dirs, depth)
+  for (const item of tree) {
+    const newPath = path === '' ? `${item.name}` : `${path}/${item.name}`
+    if (dirs || !item.isdir) result.push(newPath)
+    if (item.isdir && (currentDepth < depth || depth === -1)) {
+      const downtree = await treeToPaths(item.itemsList, newPath, dirs, depth, currentDepth + 1)
```
Comment (andrewxhill) [inline packages/buckets/src/utils.ts:55]: just changes since the buck name isn't prefixing the paths now

File: website/sidebars.json [hunk 7]
```diff
@@ -23,14 +23,15 @@
       "hub.buckets.withuserauth",
       "hub.buckets.withkeyinfo",
       "hub.buckets.gettoken",
-      "hub.buckets.getOrInit",
+      "hub.buckets.getorinit",
```
Comment (andrewxhill) [inline website/sidebars.json:26]: ooop, lowwercase only

File: packages/buckets/src/utils.ts (hunk 5 above)
Comment (carsonfarmer) [reply packages/buckets/src/utils.ts:26]: Yeh, we should just _always_ do this for external APIs. I have a linter setting coming in that'll make sure we do.

**Pull request merged**